src/sentry/static/sentry/app/components/editableText.tsx

```diff
@@ -0,0 +1,192 @@
+// eslint-disable-next-line sentry/no-react-hooks
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import TextOverflow from 'app/components/textOverflow';
+import {IconEdit} from 'app/icons/iconEdit';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import {defined} from 'app/utils';
+import useKeypress from 'app/utils/useKeyPress';
+import useOnClickOutside from 'app/utils/useOnClickOutside';
+import Input from 'app/views/settings/components/forms/controls/input';
+import Field from 'app/views/settings/components/forms/field';
+
+type Props = {
+  text: string;
+  onSetText: (label: string) => void;
+};
+
+function EditableText({text, onSetText}: Props) {
```

Comment:
We have a component similar to this already in discover. It is used for editing the name of discover saved queries. This component is more robust though. Could we update discover saved query editing to also use this component?

Reply:
sure thing... as soon as I merge this PR I can replace the discover component with this one 😉

Reply:
@priscilawebdev This is the component in question https://github.com/getsentry/sentry/blob/301a4e962ea5ae811d62becf7149762e6ada839c/src/sentry/static/sentry/app/views/eventsV2/eventInputName.tsx